ui/src/services/alertUtils.js

```diff
@@ -0,0 +1,186 @@
+        if (alerts[i].labels.severity === STATUS_WARNING) {
+          removeIndex.push(i);
+        } else if (alerts[j].labels.severity === STATUS_WARNING) {
+          removeIndex.push(j);
+        }
+      }
+    }
+  }
+
+  let removedWarningAlerts = [...alerts];
+  removeIndex.forEach((index) => removedWarningAlerts.splice(index, 1));
+
+  return removedWarningAlerts;
+};
+
+// Sort the alerts base on the `severity`
+export const sortAlerts = (alerts: AlertList): AlertList => {
+  return alerts.sort(function (a, b) {
+    return compareHealth(a.labels.severity, b.labels.severity);
+  });
+};
+
+/* 
+Format the alerts from AM
+Keep the original alert to make sure we can acces specific info in the alert list if necessary.
+Because we don't want to change the implementation because of any API changes.
+*/
+export const formatAlerts = (alerts: Array<PrometheusAlert>): AlertList => {
+  return alerts.map((alert) => {
+    return {
+      id: alert.fingerprint,
+      summary: alert.annotations.summary,
+      description: alert.annotations.description || alert.annotations.message,
+      startsAt: alert.startsAt,
+      endsAt: alert.endsAt || new Date().toISOString(),
+      severity: alert.labels.severity,
+      documentationUrl: alert.annotations.runbook_url,
+      labels: {
+        ...alert.labels,
+        parents:
+          (alert.annotations.parents && alert.annotations.parents.split(',')) ||
+          [],
+        selectors:
+          (alert.annotations.selectors &&
+            alert.annotations.selectors.split(',')) ||
+          [],
+      },
+      originalAlert: alert,
+    };
+  });
+};
+
+/* 
+This function is used when filtering the alerts, by checking if the `labels` is 
+the subset of the `filters` object. 
+Note that: The values of the selectors can be more than one, so it's possible to be an array.
+
+e.g. alertname: [NODE_FILESYSTEM_SPACE_FILLINGUP, NODE_FILESYSTEM_ALMOST_OUTOF_SPACE, 
+NODE_FILESYSTEM_FILES_FILLINGUP, NODE_FILESYSTEM_ALMOST_OUTOF_FILES]  
+*/
+export const isAlertSelected = (
+  labels: AlertLabels,
+  filters: FilterLabels,
+): boolean => {
+  return Object.getOwnPropertyNames(filters)
+    .map((key) => {
+      if (!filters[key] || !labels[key]) return false;
+      if (Array.isArray(filters[key])) {
+        if (Array.isArray(labels[key])) {
+          return (
+            filters[key].find((val) => {
+              // already check if !labels[key] return false
+              //$FlowFixMe
+              return labels[key].includes(val);
+            }) !== undefined
+          );
+        } else return filters[key].includes(labels[key]);
+      } else {
+        return labels[key].includes(filters[key]);
+      }
+    })
+    .reduce((boolA, boolB) => {
+      return boolA && boolB;
+    }, true);
+};
+
+// TODO: Call Loki Endpoint
+export const getHistoryAlerts = (
+  start: string, //TODO: maybe default last 7 days
+  end?: string = new Date().toISOString(),
+  filters: FilterLabels,
```

Comment:
IMO required parameters should be defined before optional parameters, else when you actually used the function you are being forced to pass those optional parameters (to nullish values if you want to use the default values) in order to pass your required parameters
```suggestion
  filters: FilterLabels,
  start: string, //TODO: maybe default last 7 days
  end?: string = new Date().toISOString(),
```

Reply:
I think alert fetching and filtering should be separated as in the end we are going to fetch all the alerts one time for the whole application